### PR TITLE
Guest account creation token for PayPal one-off

### PR DESCRIPTION
**Pull request by joelochlann**
We're not actually using the flashed email address anywhere, so I've swapped it out for guestAccountCreationToken since we'll need to use the same mechanism to propagate it from the redirect endpoint to the next pageview

### CODE REVIEW

File: support-frontend/app/views/newContributions.scala.html [hunk 1]
```diff
@@ -234,6 +235,10 @@ <h1 class="gu-content__blurb-header">Help us deliver the independent journalism
     window.guardian.paymentApiPayPalEndpoint = "@paymentApiPayPalEndpoint";
     window.guardian.existingPaymentOptionsEndpoint = "@existingPaymentOptionsEndpoint";
     window.guardian.csrf = { token: "@CSRF.getToken.value" };
+
+    @guestAccountCreationToken.map { guestAccountCreationToken =>
+      window.guardian.guestAccountCreationToken = "@guestAccountCreationToken";
```
Comment (jlieb10) [inline support-frontend/app/views/newContributions.scala.html]: 👍

File: support-frontend/assets/pages/new-contributions-landing/contributionsLandingInit.js [hunk 2]
```diff
@@ -218,6 +218,13 @@ const init = (store: Store<State, Action, Function>) => {
 
   initialisePaymentMethods(state, dispatch, contributionTypes);
 
+  // This will be in window.guardian if it has come from a PayPal one-off contribution,
+  // where it is returned by the Payment API to the backend, flashed into the session to preserve
+  // it through a serverside redirect, and then written into window.guardian on the thank-you page.
+  if (window.guardian.guestAccountCreationToken) {
+    setGuestAccountCreationToken(window.guardian.guestAccountCreationToken);
```
Comment (jlieb10) [inline support-frontend/assets/pages/new-contributions-landing/contributionsLandingInit.js]: 👍

**Comment by joelochlann** [comment]
This is no longer dependent on the corresponding Payment API change. I've made `guestAccountCreationToken` optional in the decoder (as it would've had to be anyway, since it's not guaranteed to be in the Payment API response).

So merging this should have no effect until the Payment API change (https://github.com/guardian/payment-api/pull/163) is merged

**Comment by joelochlann** [comment]
Have tested this in CODE, all seems fine. Can be merged once reviewed

**Pull request merged**